redfish/provider/resource_redfish_user_account.go

```diff
@@ -368,6 +369,12 @@ func (r *UserAccountResource) Update(ctx context.Context, req resource.UpdateReq
 		resp.Diagnostics.AddError("Error when retrieving accounts", "User does not exists, needs to be recreated")
 		return
 	}
+
+	// updates provider creds if username or password is changed
+	if diags = r.updateProviderServers(ctx, &plan, &state); diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
```

Comment:
We also need to check if the username and password being changed belongs to the same user given in alias, rt? Otherwise we'll just update the provider map, even if we are dealing with a completely different user using this resource.

Reply:
@rounak-adhikary , yes, that was in my codes... seems deleted after I reformat my codes. Will add that back - checking it's same user given in alias

Reply:
hi @rounak-adhikary , added